### PR TITLE
feat(invite): add `type` attribute

### PR DESCRIPTION
**Pull request by spifory**
## Summary

Implements discord/discord-api-docs#6568 by adding a `type` attribute to the `Invite` class.

## This is a **Code Change**

- [ ] I have tested my changes.
- [x] I have updated the documentation to reflect the changes.
- [x] I have run `task pyright` and fixed the relevant issues.

### CODE REVIEW

**Review by teaishealthy — APPROVED** [review]
lgtm

File: nextcord/enums.py [hunk 1]
```diff
@@ -2037,6 +2037,20 @@ class ForumLayoutType(IntEnum):
...
+class InviteType(IntEnum):
+    """Represents the type of an invite.
+
+    .. versionadded:: 3.10
+    """
+
+    guild = 0
+    """The invite is for a guild."""
+    group_dm = 1
+    """The invite is for a group DM"""
```
Comment (ArjunSharda) [inline nextcord/enums.py:2049]: Gotta correct you with one dot 😉

```suggestion
    """The invite is for a group DM."""
```

File: nextcord/enums.py [hunk 2]
```diff
@@ -2037,6 +2037,20 @@ class ForumLayoutType(IntEnum):
     """Display posts as a collection of posts with images, this is more image focused."""
 
 
+class InviteType(IntEnum):
+    """Represents the type of an invite.
+
+    .. versionadded:: 3.10
```
[the remark this answers is not in the document]
Comment (spifory) [reply nextcord/enums.py:2043]: Thanks! Best to stay ahead of time for when we do get to 3.10 /s

**Comment by EmreTech** [comment]
Related Discord Docs PR was merged.

File: nextcord/invite.py [hunk 3]
```diff
@@ -295,6 +295,9 @@ class Invite(Hashable):
         The embedded application the invite targets, if any.
 
         .. versionadded:: 2.0
+    type: :class:`int`
+        The type of invite.
+        .. versionadded:: 3.0
```
Comment (EmreTech) [inline nextcord/invite.py:300]:
```suggestion
    type: :class:`InviteType`
        The type of invite.

        .. versionadded:: 3.0
```

**Review by Skelmis — APPROVED** [review]

**Pull request merged**